### PR TITLE
fix(kafka): avoid overriding  methods in Producers and Consumers

### PR DESCRIPTION
**Pull request by mabdinur**
Resolves: https://github.com/DataDog/dd-trace-py/issues/6973

This PR achieves two goals:
 1) Fixes a bug where calling `confluent_kafka.Consumer(....).commit(offset=......)` raises the following error: `ValueError: message and offsets are mutually exclusive`. This is because `TracedConsumer`  intercepts this invocation and then calls: `Consumer(....).commit(message=None, offset=......)`.
 2) Cleans up the kafka patching code. There is no need to override a method just to call the method defined in the super class.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/kafka/patch.py [hunk 1]
```diff
@@ -230,10 +221,12 @@ def traced_commit(func, instance, args, kwargs):
         return func(*args, **kwargs)
 
     if config._data_streams_enabled:
-        message = get_argument_value(args, kwargs, 0, "message")
-        offsets = kwargs.get("offsets", [])
+        message = get_argument_value(args, kwargs, 0, "message", True)
+        # messgae and offset are mutally exclusive. Only one parameter can be passed.
```
Comment (ZStriker19) [inline ddtrace/contrib/kafka/patch.py:225]:
```suggestion
        # message and offset are mutually exclusive. Only one parameter can be passed.
```

File: releasenotes/notes/fix-kafka-commit-offsets-d86039c29e04259e.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    kafka: Fixes ``ValueError`` raised when ``Consumer.commit(offsets=...)`` is called
```
Comment (ZStriker19) [inline releasenotes/notes/fix-kafka-commit-offsets-d86039c29e04259e.yaml:4]:
```suggestion
    kafka: Fixes ``ValueError`` raised when ``Consumer.commit(offsets=...)`` is called.
```

File: ddtrace/contrib/kafka/patch.py [hunk 3]
```diff
@@ -230,10 +221,12 @@ def traced_commit(func, instance, args, kwargs):
...
         if message is not None:
             offsets = [TopicPartition(message.topic(), message.partition(), offset=message.offset())]
+        else:
+            offsets = get_argument_value(args, kwargs, 1, "offsets", True)
```
Comment (ZStriker19) [inline ddtrace/contrib/kafka/patch.py:229]: If this returns `None`, then don't we try to iterate through `None` which will throw an error?

File: releasenotes/notes/fix-kafka-commit-offsets-d86039c29e04259e.yaml (hunk 2 above)
Comment (ZStriker19) [inline releasenotes/notes/fix-kafka-commit-offsets-d86039c29e04259e.yaml:4]:
```suggestion
    kafka: Fixes ``ValueError`` raised when ``Consumer.commit(offsets=...)`` is called.
```

File: ddtrace/contrib/kafka/patch.py (hunk 1 above)
Comment (ZStriker19) [inline ddtrace/contrib/kafka/patch.py:225]:
```suggestion
        # message and offset are mutually exclusive. Only one parameter can be passed.
```

**Review by ZStriker19 — COMMENTED** [review]
nit and question

File: ddtrace/contrib/kafka/patch.py (hunk 3 above)
Comment (mabdinur) [reply ddtrace/contrib/kafka/patch.py:229]: Consumer.commit() can be called with the following arguments: `message`, `offsets` and `asynchronous`. 

The following function calls are valid:
`Consumer.commit(message=....)`
`Consumer.commit(message=..., asynchronous=....)`
`Consumer.commit(offsets=None, asynchronous=...)`
`Consumer.commit(asynchronous=...)`

The following function calls are INVALID. This raises a ValueError:
`Consumer.commit(message=...., offsets=.....)`


There is a scenario where both message and offsets are not passed as keyword arguments. In this case `get_argument_value(....)` should not raise an ArgumentError. It should just return `None`.



Docs for reference: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.commit

**Comment by ZStriker19** [comment]
Kafka test is failing it seems

**Comment by mabdinur** [comment]
> Kafka test is failing it seems

It's failing because I forgot to push up a new snapshot file

We should be good

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**